File: Python3/프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    flag = True
    answer = 0
    while flag == True:
        answer += 1
        flag = False
        for i in range(len(arr)):
            if 50 <= arr[i] and arr[i] % 2 == 0:
                flag = True
                arr[i] /= 2
            elif arr[i] < 50 and arr[i] % 2 == 1:
                flag = True
                arr[i] = arr[i] * 2 + 1
                
        
    return answer - 1

File: C++/프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;
    bool flag = true;
    while(flag) {
        flag = false;
        answer++;
        for(int i = 0; i < arr.size(); i++) {
            if(50 <= arr[i] && arr[i] % 2 == 0) {
                flag = true;
                arr[i] /= 2;
            } else if(arr[i] < 50 && arr[i] % 2 == 1) {
                flag = true;
                arr[i] *= 2;
                arr[i]++;
            }
        }
    }
    return answer - 1;
}